_tutorials/t04_renv.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: tutorial
+title: Managing renv
+shortID: renv
+lastUpdated: 2022-11-24
+model: MAgPIE
+modelVersion: 4.6.0
+author: pf
+level: 3
+requirements:
+  - R installed (<https://www.r-project.org/>)
+  - MAgPIE model folder cloned
+lessonsContent:
+  - Understand what renv is, how it works and why and how MAgPIE uses it.
+  - Update and restore renvs via piamenv functions.
+  - Get an overview of renv functions to get more control.
+categories: tutorial
+published: true
+---
+# basics
+MAgPIE uses [renv](https://rstudio.github.io/renv/) for managing required R packages. When starting R from the MAgPIE folder the corresponding renv is automatically activated. This means that packages are installed to and loaded from the renv subfolder instead of the usual package library. Thus updates in the global package library do not affect MAgPIE which is especially important on multi-user systems where packages may be loaded by some users while others are updating these packages. In such a situation the loaded packages are no longer valid and R crashes. Another advantage of renv is that it can document exactly which packages are currently installed, and it makes it easy to go back to a previous package configuration via renv.lock files.
```

Comment:
```suggestion
MAgPIE uses [renv](https://rstudio.github.io/renv/) for managing required R packages. When starting R from the MAgPIE folder the corresponding renv is automatically activated. This means that packages are installed to and loaded from the renv subfolder instead of the usual package library. Thus, updates in the global package library do not affect MAgPIE. This is especially important on multi-user systems where packages may be loaded by some users while others are updating these packages. In such a situation, the loaded packages are no longer valid and R crashes. Another advantage of renv is that it can document exactly which packages are currently installed, and it makes it easy to go back to a previous package configuration via renv.lock files.
```